blockware/lib/Colors/Colors.h

```diff
@@ -27,4 +35,116 @@ static inline int randomColor()
   return COLORS[colorIndex];
 }
 
+static inline int randomLogoColor()
+{
+  // random logo color
+  const int COLORS[] = {LOGO_BLUE, LOGO_GREEN, LOGO_ORANGE, LOGO_PURPLE, LOGO_RED, LOGO_YELLOW};
+  int colorIndex = floor((sizeof(COLORS)/sizeof(int)) * randomf());
+  return COLORS[colorIndex];
+}
+
+// Takes a 5-6-5 RGB and returns the red component
+static uint8_t getRedRaw(uint16_t color)
```

Comment:
This is breaking the build, I think you need to include something to bring in these types. Is it maybe `#include <inttypes.h>`?

Reply:
> Would you mind just replacing the `fireworks` blockware with your version? Then it will be easier to see what's changed, and I think your version is definitively better than the existing version.

Happy to, will get that up in a bit.